### DISCUSSION TITLE
Git Commits Showing Another Users Name

### DISCUSSION THREAD

**Discussion by aetherrealm** [community/community#55259] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Bug

### Body

I created a private repo on my profile this afternoon and did my initial commit. Strangely, on the repo page, and if I go to commit history, it shows a different user, someone who I've never met and doesn't appear to really be an active user. If I go to my profile activity it shows that I created the project, but the new commits do not appear in my activity. If I go to the other users profile it doesn't show any activity, but I suppose they could have that setting to display private repo contributions turned off?

Initially I thought it might be some sort of cosmetic bug, but the fact that my profile contributions aren't changing makes me think there's actually some keying issues going on in the backend.

Kind of sketchy.

![image](https://github.com/community/community/assets/52265137/112c5f59-5b4f-42da-a4de-6202f3cfb4e3)
![image](https://github.com/community/community/assets/52265137/48f085e9-9911-4f25-9f5e-b4e6a20c766a)
![image](https://github.com/community/community/assets/52265137/b37a2259-ab0d-43cd-8336-e30671570b76)
![image](https://github.com/community/community/assets/52265137/c56a2e7d-140c-4111-b2b6-71b4b164f75c)

**Comment by airtower-luna** [comment]
GitHub assigns commits to users based on the *author email* recorded in the commit. Most likely there's a mistake in your `user.email` setting on the computer where you made the commits. You can use `git log` or apply `.patch` to the URL of a commit on GitHub to check the recorded address.

Make sure your [user name and email are set correctly](https://git-scm.com/book/en/v2/Getting-Started-First-Time-Git-Setup#_your_identity), for GitHub to recognize your commits `user.email` must be an address connected to your account. That only changes the author email for future commits, see [Git Tools - Rewriting History](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History) for how to fix past ones.
